### PR TITLE
feat: add download manifests command

### PR DESCRIPTION
**Pull request by ponces**
- Adds a new `Download Manifests` command:
    - Example: `cmf-portal download-manifests --name testenv --output /tmp/output`
- Uses existing code from the deploy commands (`EnvironmentDeploymentHandler` and `AppInstallationHandler`)
- Reuses moved code to prevent code duplication
- Bumps the version of `upload-artifact` Github action. v2 is deprecated

### CODE REVIEW

**Comment by afjsilva** [comment]
Please, also, add to README.md an usage example and explain the command added like the other examples :)

**Comment by ponces** [comment]
> Please, also, add to README.md an usage example and explain the command added like the other examples :)

README updated :)

File: src/Common/Services/ArtifactDownloaderHandler.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            }
+            else
+            {
+                // Download the attachment
+                _session.LogDebug($"Downloading attachment {attachmentToDownload.Filename}");
+
+                string outputFile = "";
+                using (DownloadAttachmentStreamingOutput downloadAttachmentOutput = await new DownloadAttachmentStreamingInput() { attachmentId = attachmentToDownload.Id }.DownloadAttachmentAsync(true))
+                {
+                    int bytesToRead = 10000;
```
[the remark this answers is not in the document]
Comment (ponces) [reply src/Common/Services/ArtifactDownloaderHandler.cs]: I opted to use an EntityBase object as an argument and use its type to understand what we are handling. 
Let me know what you think :)

File: src/Common/Handlers/DownloadDeployArtifactHandler.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                string errorMessage = $"Customer environment {customerEnvironmentName} doesn't exist...";
+                _session.LogInformation(errorMessage);
+
+                throw new NotFoundException(errorMessage);
+            }
+
+            // start download
+            _session.LogInformation($"Downloading manifests for Customer environment {customerEnvironmentName}...");
+            await _manifestsDownloaderHandler.Handle(environment, outputDir);
+            _session.LogInformation($"Manifests successfully download to {outputDir.FullName}.");
```
[the remark this answers is not in the document]
Comment (CidHighwindVII) [reply src/Common/Handlers/DownloadDeployArtifactHandler.cs:50]: Note: For this command, outputDir has value.

deploy -params \"c:\\PortalSDKTests\\params.json\" --verbose -lic \"CMF - PT_Development_v11.0.0_ED20250810\" --site \"CMF - PT\" --package=\"@criticalmanufacturing\\mes:11.0.4\" --target \"OpenShiftOnPremisesTarget\" --output \"c:PortalSDKTests\" --deploymentTimeoutMinutes 60

[the remark this answers is not in the document]
Comment (ponces) [reply src/Common/Handlers/DownloadDeployArtifactHandler.cs:50]: Async functions can't have ref arguments. I'll deal with this by ensuring output path is not null before calling the handler.

**Review by CidHighwindVII — APPROVED** [review]

**Review by CidHighwindVII — APPROVED** [review]

**Review by CidHighwindVII — APPROVED** [review]

**Review by afjsilva — COMMENTED** [review]
@ponces Can you do the version bump on the "Version.props" file for 1.14.11?

**Comment by afjsilva** [comment]
@ponces Thanks for the version bump.
Other think, and I'm sorry for this, but after analyze and discuss with Carlos, we think that the name "Manifest" in the handlers don't match the final purpose, because in our context a manifest match XML files and this can let some confusion in the team development. Can you replace for "Artifact"? Because we are currently the entity attachments that are in this case most of the times, yamls and other files. The aggregation of this files we call artifacts.

**Review by afjsilva — APPROVED** [review]

**Pull request merged**